controllers/v2/unavailability/api.py

```diff
@@ -53,16 +59,21 @@ def delete(self, user_id, event_id):
 
 class VolunteerUnavailabilityV2(Resource):
 
+    def __init__(self, event_repository):
+        self.event_repository = event_repository
+
     @requires_auth
     @marshal_with(volunteer_unavailability_time)
     @is_user_or_has_role(None, UserType.ROOT_ADMIN)
     def get(self, user_id):
         with session_scope() as session:
-            volunteer_unavailability_record = fetch_event(session, user_id)
+            event_repository = EventRepository(session)
```

Comment:
Add it though the *constructor* of the class, otherwise this is correct

Reply:
Hi Ben, seems that if I use the session_scope, I can only instantiate the EventRepository individually within each request method in api.py. I'm not really sure how can it though the constructor of the class. I test these four requests in Postman and seems they all work normally now. Do I still need to do some changes for it?

Reply:
My changes weren't about functionality, but code maintainability. Passing EventRepository in through the constructor means we can unit test the endpoints individually. I've put instructions in my feedback on how to make the changes I requested, but message me if you need further help.